app/src/main/java/com/x8bit/bitwarden/data/autofill/fido2/processor/Fido2ProviderProcessorImpl.kt

```diff
@@ -201,24 +203,37 @@ class Fido2ProviderProcessorImpl(
                         .getPasskeyAssertionOptionsOrNull(requestJson = option.requestJson)
                         ?.relyingPartyId
                         ?: throw GetCredentialUnknownException("Invalid data.")
-
-                    vaultRepository
-                        .silentlyDiscoverCredentials(
-                            userId = userId,
-                            fido2CredentialStore = fido2CredentialStore,
-                            relyingPartyId = relyingPartyId,
-                        )
-                        .fold(
-                            onSuccess = { it.toCredentialEntries(option) },
-                            onFailure = {
-                                throw GetCredentialUnknownException("Error decrypting credentials.")
-                            },
-                        )
+                    buildCredentialEntries(relyingPartyId, option)
                 } else {
                     throw GetCredentialUnsupportedException("Unsupported option.")
                 }
             }
 
+    private suspend fun buildCredentialEntries(
+        relyingPartyId: String,
+        option: BeginGetPublicKeyCredentialOption,
+    ): List<CredentialEntry> {
+        val cipherViews = vaultRepository
+            .ciphersStateFlow
+            .value
+            .data
+            ?.filter { it.isActiveWithFido2Credentials }
+            ?: emptyList()
+        val result = vaultRepository
+            .getDecryptedFido2CredentialAutofillViews(cipherViews)
```

Comment:
So this is just a temporary fix until the sdk fixes `silentlyDiscoverCredentials`, right?

Reply:
Correct.